### PR TITLE
[CRIMAPP-975] self employed schema

### PR DESCRIPTION
**Pull request by hiboabd**
## Description of change
Proposed schema for self employed flow 
Yet to add to the maat json, this is pending a conversation with maat side 

Question: is it worth having a separate fixture for self employed?

## Link to relevant ticket
[CRIMAPP-975](https://dsdmoj.atlassian.net/browse/CRIMAPP-975)

## Additional notes

[CRIMAPP-975]: https://dsdmoj.atlassian.net/browse/CRIMAPP-975?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: schemas/1.0/general/business.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "trading_start_date": { "type": ["string", "null"], "format": "date" },
+    "address": { "type": ["object", "null"]},
+    "has_additional_owners": { "type": ["string", "null"], "enum": ["yes", "no", null] },
+    "additional_owners": { "type": ["string", "null"] },
+    "has_employees": { "type": ["string", "null"], "enum": ["yes", "no", null] },
+    "number_of_employees": { "type": ["integer", "null"] },
+    "salary": { "type": ["number", "null"] },
+    "total_income_share_sales": { "type": ["number", "null"] },
+    "percentage_profit_share": { "type": ["number", "null"] },
+    "turnover": { "type": ["object", "null"]},
```
Comment (timpeat) [inline schemas/1.0/general/business.json:21]: Not required, but the schema for turnover, drawings etc could be made more specific by referencing the definition for amount. Same could be done for address.

**Review by timpeat — COMMENTED** [review]
@hiboabd are you able to add a general definition for 'amount' (like we have for address) and then reference it for the values that use it.

i.e.     "turnover": { "$ref": "general/amount.json" },

otherwise, looks great.

File: lib/laa_crime_schemas/types/types.rb [hunk 2]
```diff
@@ -287,6 +287,8 @@ module Types
                                          no_check_required
                                          checker_unavailable
                                          confirmed])
+
+    BusinessType = String.enum(*%w[self_employed partnership director_or_shareholder])
```
Comment (arthurashman) [inline lib/laa_crime_schemas/types/types.rb:291]: Any thoughts on naming this BusinessInvolvementType?

Comment (arthurashman) [reply lib/laa_crime_schemas/types/types.rb:291]: Hmm, I've just seen Tim's PR- didn't realise this was the page name/in the designs. I'm not a fan. But that's fine

Comment (hiboabd) [reply lib/laa_crime_schemas/types/types.rb:291]: Happy to change to business involvement type - i see how that's a lot more accurate

**Review by timpeat — APPROVED** [review]
💯 thank you!

**Pull request merged**